warehouse/query-core/src/test/java/datawave/query/util/FuzzyAttributeComparatorTest.java

```diff
@@ -0,0 +1,142 @@
+package datawave.query.util;
+
+import com.google.common.collect.Sets;
+import datawave.query.attributes.Attribute;
+import datawave.query.attributes.Attributes;
+import datawave.query.attributes.Content;
+import datawave.query.attributes.IpAddress;
+import datawave.query.attributes.PreNormalizedAttribute;
+import datawave.query.attributes.PreNormalizedAttributeFactory;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.hadoop.io.Text;
+import org.hamcrest.CoreMatchers;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class FuzzyAttributeComparatorTest {
```

Comment:
This test looks fine overall, but let's add a `DocumentTest` and roll through a number of expected cases.
1. add two instances of `Attribute` to document
2. add two instances of `Attributes` to document
3. add a mix of `Attribute` and `Attributes` to document

For each case outlined above hit the following variants
- test case without duplicate values
- test case with duplicate value
- test case with duplicate value in different Attribute types (Content vs. PreNormalizedAttribute)

Also be sure to hit a few edge cases
- merge attribute with document key vs. no document key
- merge attribute with toKeep flag set/not set
- merge attribute with different visibilities (test will act as documentation of expected behavior)

Reply:
Updated merging logic and added new edge cases as requested